packages/js/components/src/table/index.tsx

```diff
@@ -0,0 +1,250 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import classnames from 'classnames';
+import { createElement, Fragment, useState } from '@wordpress/element';
+import { find, first, without } from 'lodash';
+import React from 'react';
+import {
+	Card,
+	CardBody,
+	CardFooter,
+	CardHeader,
+	// @ts-expect-error: Suppressing Module '"@wordpress/components"' has no exported member '__experimentalText'
+	__experimentalText as Text,
+} from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import EllipsisMenu from '../ellipsis-menu';
+import MenuItem from '../ellipsis-menu/menu-item';
+import MenuTitle from '../ellipsis-menu/menu-title';
+import Pagination from '../pagination';
+import Table from './table';
+import TablePlaceholder from './placeholder';
+import TableSummary, { TableSummaryPlaceholder } from './summary';
+import { TableCardProps } from './types';
+
+const defaultOnQueryChange =
+	( p: string ) =>
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	( ...props: any ) => {};
+
+const defaultOnColumnsChange = (
+	showCols: Array< string >,
+	key?: string
+) => {};
+/**
+ * This is an accessible, sortable, and scrollable table for displaying tabular data (like revenue and other analytics data).
+ * It accepts `headers` for column headers, and `rows` for the table content.
+ * `rowHeader` can be used to define the index of the row header (or false if no header).
+ *
+ * `TableCard` serves as Card wrapper & contains a card header, `<Table />`, `<TableSummary />`, and `<Pagination />`.
+ * This includes filtering and comparison functionality for report pages.
+ */
+const TableCard: React.VFC< TableCardProps > = ( {
+	actions,
+	className,
+	hasSearch,
+	headers = [],
+	ids,
+	isLoading = false,
+	onQueryChange = defaultOnQueryChange,
+	onColumnsChange = defaultOnColumnsChange,
+	onSort,
+	query = {},
+	rowHeader = 0,
+	rows = [],
+	rowsPerPage,
+	showMenu = true,
+	summary,
+	title,
+	totalRows,
+	rowKey,
+	emptyMessage = undefined,
+	...props
+} ) => {
+	// eslint-disable-next-line no-console
+	const getShowCols = ( _headers: TableCardProps[ 'headers' ] = [] ) => {
+		return _headers
+			.map( ( { key, visible } ) => {
+				if ( typeof visible === 'undefined' || visible ) {
+					return key;
+				}
+				return false;
+			} )
+			.filter( Boolean );
```

Comment:
If I'm not mistaken,  `getShowCols` only returns the `string[].` 

If we can change the type here, we don't need `as string[];` in line 97 and line 101. I think it should be more reasonable that showCols type is string[], not (string | false)[].

```suggestion
			.filter( Boolean ) as string[];
```

Reply:
Good suggestion 👍